### PR TITLE
7 set up frontend testing framework

### PR DESCRIPTION
**Pull request by sellsol**
### Description
Set up a testing directory for frontend unit tests, and add a testing process into the Github workflow so that code that doesn't pass these tests cannot be merged into main. Write one or two very basic ones just to be sure they work (at your discretion).  

### To-do
- [x] Set up frontend testing framework
- [x] Write one or two basic unit tests
- [x] Integrate into Github workflow
- [x] Update Readme for documentation

### Notes
* The file name of the backend tests have changed so there's a new workflow in GitHub Actions under the same name. Will just have to delete the old one from GitHub once all the branches are migrated over
* Only way to get tsconfig to work properly from the root directory of the repo is to have one in the root directory. I'm not sure what the other tsconfigs do, if they break because of this change let me know. But this was what I had to do to get paths working in tests.  
* Playwright config is autogenerated
* Changed the sidepane header text to only show on v-if, for easier testing - honestly this was what was supposed to be done. That file will need a little rewrite later to decouple the sidepane and the button so we can just not render the sidepane. Out of scope for this ticket. 
* Vitest is needed for unit tests, Playwright is needed for e2e tests. Can't really use one for the other

### CODE REVIEW

File: src/client/tsconfig.json [hunk 1]
```diff
@@ -1,15 +1,3 @@
...
-      "path": "./tsconfig.app.json"
-    }
-  ],
-  "compilerOptions": {
-    "forceConsistentCasingInFileNames": true,
-    "skipLibCheck": true
-  }
-}
\ No newline at end of file
+  "extends": "../../tsconfig.base.json"
```
Comment (keyman015) [inline src/client/tsconfig.json:2]: I feel the file should be changed to

```
{
  "extends": "./../../tsconfig.base.json",
  "files": [],
  "references": [
    { "path": "./tsconfig.node.json" },
    { "path": "./tsconfig.app.json" }
  ]
}
```
to ensure nothing breaks.

File: README.md [hunk 2]
```diff
@@ -142,7 +142,17 @@ pytest
...
-🚧 Client testing setup is currently under development. 🚧
+Tests for the Vue client are written with Vitest for unit tests, and Playwright for end-to-end tests, the commands for which have been added as `npm` scripts.
+
+Run in the root directory (unit / e2e depending on what is wanted):
+
+```
+cd src/client
+npm run test:unit
+npm run test:e2e
+```
```
Comment (keyman015) [inline README.md:153]: Should the user be warned to install play-wright and its dependencies as thats separate from the main packages?

**Review by keyman015 — CHANGES_REQUESTED** [review]
This looks amazing, however just some minor comments about documentation and configuration files.

File: README.md (hunk 2 above)
Comment (sellsol) [reply README.md:153]: Playwright and dependencies get installed on `npm i` right? We have that earlier on. Let me know if not

Comment (keyman015) [reply README.md:153]: for me that was a separate install command i had to run, however it did prompt me properly when i tried to run it pretty sure

Comment (keyman015) [reply README.md:153]: Im happy to mark this as resolved.

**Review by keyman015 — APPROVED** [review]
LGTM

**Pull request merged**